### PR TITLE
[SPARK-25884][SQL] Add TBLPROPERTIES and COMMENT, and use LOCATION when SHOW CREATE TABLE.

### PR DESCRIPTION
**Pull request by ueshin**
## What changes were proposed in this pull request?

When `SHOW CREATE TABLE` for Datasource tables, we are missing `TBLPROPERTIES` and `COMMENT`, and we should use `LOCATION` instead of path in `OPTION`.

## How was this patch tested?

Splitted `ShowCreateTableSuite` to confirm to work with both `InMemoryCatalog` and `HiveExternalCatalog`, and  added some tests.

### CODE REVIEW

**Comment by ueshin** [comment]
cc @cloud-fan

File: sql/core/src/test/resources/sample.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{"a" : "2" ,"b" : "blah", "c_!@(3)":1}
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sample.json]: why add this file?

Comment (ueshin) [reply sql/core/src/test/resources/sample.json]: It is needed to run a test "data source table with user specified schema". Actually we just needed to move the file. I'll remove the extra one.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveShowCreateTableSuite.scala [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+             |)
+           """.stripMargin
+        )
+
+        checkCreateTable("t1")
+      }
+    }
+  }
+
+  test("partitioned hive table") {
```
Comment (cloud-fan) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveShowCreateTableSuite.scala]: do we have tests for partitioned/bucketed data source table?

**Comment by cloud-fan** [comment]
LGTM except some minor comments

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by srowen** [comment]
@cloud-fan @ueshin I'm not sure, but I'm seeing this failure regularly in master builds and I wonder if this could be the cause? in two builds it started failing with this (and another) change:

https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-maven-hadoop-2.7/5561/testReport/
https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-master-test-maven-hadoop-2.7-ubuntu-scala-2.12/574/testReport/

**Comment by ueshin** [comment]
Seems like we still need `sample.json` in hive module. I'll submit a follow-up pr.

**Pull request closed without merging**